src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3687,8 +3687,14 @@ private IEnumerable<SourceRepository> GetEffectiveSources(IEnumerable<SourceRepo
                 PackagesFolderSourceRepository
             };
 
-            effectiveSources.AddRange(primarySources);
-            effectiveSources.AddRange(secondarySources);
+            if (primarySources != null)
```

Comment:
A few lines above, we are using `primarySources?.Count()` to check against null failures, but this code doesn't follow that. We should either remove the `?.` above, or add guards here.